pyaerocom/aeroval/obsentry.py

```diff
@@ -1,29 +1,58 @@
 import logging
+from pathlib import Path
 from traceback import format_exc
+from typing import Literal
+
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    field_validator,
+    model_validator,
+)
 
 from pyaerocom import const
-from pyaerocom._lowlevel_helpers import BrowseDict, ListOfStrings, StrType
+from pyaerocom._lowlevel_helpers import LayerLimits
 from pyaerocom.exceptions import InitialisationError
-from pyaerocom.metastandards import DataSource
 
 logger = logging.getLogger(__name__)
 
 
-class ObsEntry(BrowseDict):
-    """Observation configuration for evaluation (dictionary)
+SUPPORTED_VERT_CODES: tuple[
+    str,
+    str,
+    str,
+] = (
+    "Column",
+    "Profile",
+    "Surface",
+)
+
+ALT_NAMES_VERT_CODES: dict = dict(ModelLevel="Profile")
+
+
+SUPPORTED_VERT_LOCS: tuple[str, str, str] = (
+    "ground",
+    "space",
+    "airborne",
+)
+
+
+class ObsEntry(BaseModel):
+    """Observation configuration for evaluation (BaseModel)
```

Comment:
Generally, I don't like it if the interface changes (here access to all attributes from ["XXX"] to .XXX) This might destroy existing code. Ok since obsentry is mostly internal.

Reply:
Agree that where the user interacts with the code, we should try to minimize the changes to the interface. But also I think it's hard to provide structure to the code if everything is fundamentally a `BrowseDict` without everything becoming very nested. There's a balance there, and I think this is a good place to move away from the older dictionary-style accessing.